realm/realm-library/src/androidTest/java/io/realm/ObjectChangeSetTests.java

```diff
@@ -0,0 +1,281 @@
+        allTypes.addChangeListener(new RealmObjectChangeListener<RealmModel>() {
+            @Override
+            public void onChange(RealmModel object, ObjectChangeSet changeSet) {
+                assertEquals(fieldNames.length, changeSet.getChangedFields().length);
+                for (String name : fieldNames) {
+                    for (String field : changeSet.getChangedFields()) {
+                        if (field.equals(name)) {
+                            break;
+                        }
+                        fail("Cannot find field " + name + " in field changes.");
+                    }
+                }
+                looperThread.testComplete();
+            }
+        });
+        looperThread.keepStrongReference.add(allTypes);
+    }
+
+    private void listenerShouldNotBeCalled(AllTypes allTypes) {
+        allTypes.addChangeListener(new RealmObjectChangeListener<RealmModel>() {
+            @Override
+            public void onChange(RealmModel object, ObjectChangeSet changeSet) {
+                fail();
+            }
+        });
+        looperThread.postRunnableDelayed(new Runnable() {
+            @Override
+            public void run() {
+                looperThread.testComplete();
+            }
+        }, 100);
+    }
+
+    @Test
+    @RunTestInLooperThread(before = PopulateOneAllTypes.class)
+    public void objectDeleted() {
+        Realm realm = looperThread.realm;
+        AllTypes allTypes = realm.where(AllTypes.class).findFirst();
+        checkDeleted(allTypes);
+        realm.beginTransaction();
+        allTypes.deleteFromRealm();
+        realm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread(before = PopulateOneAllTypes.class)
+    public void changeIntField() {
+        Realm realm = looperThread.realm;
+        AllTypes allTypes = realm.where(AllTypes.class).findFirst();
+        checkChangedField(allTypes, AllTypes.FIELD_LONG);
+        realm.beginTransaction();
+        allTypes.setColumnLong(42);
+        realm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread(before = PopulateOneAllTypes.class)
+    public void changeStringField() {
+        Realm realm = looperThread.realm;
+        AllTypes allTypes = realm.where(AllTypes.class).findFirst();
+        checkChangedField(allTypes, AllTypes.FIELD_STRING);
+        realm.beginTransaction();
+        allTypes.setColumnString("42");
+        realm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread(before = PopulateOneAllTypes.class)
+    public void changeFloatField() {
+        Realm realm = looperThread.realm;
+        AllTypes allTypes = realm.where(AllTypes.class).findFirst();
+        checkChangedField(allTypes, AllTypes.FIELD_FLOAT);
+        realm.beginTransaction();
+        allTypes.setColumnFloat(42.0f);
+        realm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread(before = PopulateOneAllTypes.class)
+    public void changeDoubleField() {
```

Comment:
All types doesn't seem to be covered. Missing is at least: `boolean`, `float`, `byte[]`, `Date`